addons/arsenal/functions/fnc_arsenalModule.sqf

```diff
@@ -0,0 +1,68 @@
+#include "script_component.hpp"
+/*
+ * Author: veteran29
+ * Initialization function of Unit Arsenal module.
+ *
+ * Arguments:
+ * 0: Logic <OBJECT>
+ *
+ * Return Value:
+ * None
+ *
+ * Public: No
+ */
+
+if (!IS_MOD_LOADED(ace_arsenal)) exitWith {
+    ERROR("Can not add arsenal, ace_arsenal is not loaded");
+};
+
+[{
+    params [
+        ["_logic", objNull, [objNull]]
+    ];
+
+    TRACE_1("Initializing arsenal module",_logic);
+
+    private _deleteUnits = _logic getVariable [QGVAR(deleteUnits), true];
+    private _playerItems = _logic getVariable [QGVAR(playerItems), true];
+    private _allGoggles = _logic getVariable [QGVAR(allGoggles), true];
+    private _categories = _logic getVariable [QGVAR(categories), ATTRIBUTE_CATEGORIES_DEFAULT_VALUE];
+    TRACE_4("Module attributes",_deleteUnits,_playerItems,_allGoggles,_categories);
+
+    private _syncedObjects = synchronizedObjects _logic;
+    private _syncedUnits = _syncedObjects select {_x isKindOf "CAManBase"};
+    private _syncedTriggers = _syncedObjects select {_x isKindOf "EmptyDetector"};
+    private _syncedArsenals = _syncedObjects - _syncedUnits - _syncedTriggers;
+    TRACE_3("Module synced",_syncedUnits,_syncedTriggers,_syncedArsenals);
+
```

Comment:
It would be nice to show some warning or something if no units/objects are synchronised as someone may forgot or accidentally remove the connection.

Also I guess the module should fail and not delete the units if `_syncedArsenals` is empty.